opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -1238,4 +1239,25 @@ shouldIterateNetwork(const int reportStepIdx,
            network_imbalance > balance.pressure_tolerance();
 }
 
+std::vector<int>
+BlackoilWellModelGeneric::
+getCellsForConnections(const Well& well) const
+{
+    std::vector<int> wellCells;
+    // All possible connections of the well
+    const auto& connectionSet = well.getConnections();
+    wellCells.reserve(connectionSet.size());
+
+    for (std::size_t c = 0; c < connectionSet.size(); c++)
+    {
+        const auto& connection = connectionSet.get(c);
+        int compressed_idx = compressedIndexForInterior(connection.global_index());
+        if ( compressed_idx >= 0 ) { // Ignore connections in inactive/remote cells.
+            wellCells.push_back(compressed_idx);
+        }
+    }
```

Comment:
You can use range-for here since the `WellConnections` class has the requisite `begin()` and `end()` member functions.

Reply:
done.